File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string ret;
        int i = 0;
        while (i < word1.length() && i < word2.length()) {
            ret.push_back(word1[i]);
            ret.push_back(word2[i]);
            ++i;
        }
        if (i < word1.length()) {
            while (i < word1.length()) {
                ret.push_back(word1[i]);
                ++i;
            }
        }
        else if (i < word2.length()) {
            while (i < word2.length()) {
                ret.push_back(word2[i]);
                ++i;
            }
        }
        return ret;
    }
};

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        mergedString = ""
        minLength = min(len(word1), len(word2))
        i = 0
        while i < minLength:
            mergedString += word1[i]
            mergedString += word2[i]
            i += 1
            
        if (len(word1) > len(word2)):
            mergedString += word1[i:]
            
        if (len(word2) > len(word1)):
            mergedString += word2[i:]
            
        return mergedString
        